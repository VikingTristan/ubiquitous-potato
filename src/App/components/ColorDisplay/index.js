import React from "react";
import PropTypes from "prop-types";
import ColorPreview from "@components/ColorPreview";

const ColorDisplay = ({ id, colorList, head, description, subListNames }) => (
    <>
        <h2 id={id}>{head}</h2>
        {description}

        {subListNames ? subListNames.map((subListName, index) => (
            <div key={index}>
                <h3>{subListName}</h3>
                <ul className="list color-preview">
                    {colorList.filter(elem => elem.inSubLists.includes(subListName)).map((elem, i) => {
                        /* eslint-disable */
                        const { inSubLists, ...elemProps } = elem; // Exclude property not needed by ColorPreview
                        /* eslint-enable */

                        return (
                            <li key={i} >
                                <ColorPreview {...elemProps} />
                            </li>
                        );
                    })}
                </ul>
            </div>))
            :
            <ul className="list color-preview">
                {colorList.map((elem, i) => {
                    /* eslint-disable */
                    const { inSubLists, ...elemProps } = elem; // Exclude property not needed by ColorPreview
                    /* eslint-enable */

                    return (
                        <li key={i} >
                            <ColorPreview {...elemProps} />
                        </li>
                    );
                })}
            </ul>
        }
    </>
);

ColorDisplay.propTypes = {
    id: PropTypes.string.isRequired,
    colorList: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string.isRequired,
        hex: PropTypes.string.isRequired,
        subListNames: PropTypes.arrayOf(PropTypes.string)
    })).isRequired,
    head: PropTypes.string.isRequired,
    description: PropTypes.any,
    subLists: PropTypes.bool
};

export default ColorDisplay;
