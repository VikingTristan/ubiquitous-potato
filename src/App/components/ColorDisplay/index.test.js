import React from "react";
import { shallow } from "enzyme";

import ColorDisplay from "./index";

describe("Component: ColorDisplay", () => {
    it("is defined", () => {
        expect(ColorDisplay).toBeDefined();
    });

    it("renders ColorDisplay", () => {
        const mockColor = {
            name: "Yellow",
            hex: "#FDC129"
        };

        const mockColorDisplayElement = {
            id: "test",
            colorList: [mockColor, mockColor],
            head: "Test Head",
            description: <p>This is a description</p>
        };

        const wrapper = shallow(<ColorDisplay {...mockColorDisplayElement}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.html()).toContain("<h2 id=\"test\">Test Head</h2>");
        expect(wrapper.html()).toContain("<p>This is a description</p>");
        expect(wrapper.html()).toContain("<small>Hex color code: #FDC129</small>");
    });

    it("renders colorDisplay with subLists when subListNames provided", () => {
        const mockColor1 = {
            name: "Yellow",
            hex: "#FDC129",
            inSubLists: ["Communication color"]
        };
        const mockColor2 = {
            name: "Brown",
            hex: "#512B2B",
            inSubLists: ["Text color"]
        };
        const mockColor3 = {
            name: "Brown",
            hex: "#512B2B",
            inSubLists: ["Text color"]
        };

        const mockColorDisplayElement = {
            id: "test",
            colorList: [mockColor1, mockColor2, mockColor3],
            head: "Test Head",
            description: <p>This is a description</p>,
            subListNames: ["Communication color", "Text color"]
        };

        const wrapper = shallow(<ColorDisplay {...mockColorDisplayElement} />);
        const colorLists = wrapper.find(".list.color-preview");

        expect(wrapper).toMatchSnapshot();
        expect(colorLists.length).toEqual(mockColorDisplayElement.subListNames.length);
        expect(wrapper.contains(<h3>Communication color</h3>)).toEqual(true);
        expect(wrapper.contains(<h3>Text color</h3>)).toEqual(true);
        expect(wrapper.html()).toContain("<small>Hex color code: #FDC129</small>");
        expect(wrapper.html()).toContain("<small>Hex color code: #512B2B</small>");
    });
});
