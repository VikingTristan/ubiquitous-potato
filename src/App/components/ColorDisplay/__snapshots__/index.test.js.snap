// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ColorDisplay renders ColorDisplay 1`] = `
<Fragment>
  <h2
    id="test"
  >
    Test Head
  </h2>
  <p>
    This is a description
  </p>
  <ul
    className="list color-preview"
  >
    <li
      key="0"
    >
      <ColorPreview
        hex="#FDC129"
        name="Yellow"
      />
    </li>
    <li
      key="1"
    >
      <ColorPreview
        hex="#FDC129"
        name="Yellow"
      />
    </li>
  </ul>
</Fragment>
`;

exports[`Component: ColorDisplay renders colorDisplay with subLists when subListNames provided 1`] = `
<Fragment>
  <h2
    id="test"
  >
    Test Head
  </h2>
  <p>
    This is a description
  </p>
  <div
    key="0"
  >
    <h3>
      Communication color
    </h3>
    <ul
      className="list color-preview"
    >
      <li
        key="0"
      >
        <ColorPreview
          hex="#FDC129"
          name="Yellow"
        />
      </li>
    </ul>
  </div>
  <div
    key="1"
  >
    <h3>
      Text color
    </h3>
    <ul
      className="list color-preview"
    >
      <li
        key="0"
      >
        <ColorPreview
          hex="#512B2B"
          name="Brown"
        />
      </li>
      <li
        key="1"
      >
        <ColorPreview
          hex="#512B2B"
          name="Brown"
        />
      </li>
    </ul>
  </div>
</Fragment>
`;
